aptos-move/aptos-aggregator/src/bounded_math.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright © Aptos Foundation
+
+use aptos_logger::error;
+use move_binary_format::errors::{PartialVMError, PartialVMResult};
+use move_core_types::vm_status::StatusCode;
+
+// When bounded math operation overflows
+// Generally means addition exceeded limit.
+pub(crate) const EBOUND_OVERFLOW: u64 = 0x02_0001;
+
+/// When bounded math operation underflows
+/// Generally means subtraction went below 0.
+pub(crate) const EBOUND_UNDERFLOW: u64 = 0x02_0002;
+
+/// When updating the aggregator start value (due to read operations
+/// or at the end of the transaction), we realize that mistakenly raised
+/// an overflow in one of the previus try_add operation.
+pub(crate) const EEXPECTED_OVERFLOW: u64 = 0x02_0003;
+
+/// When updating the aggregator start value (due to read operations
+/// or at the end of the transaction), we realize that mistakenly raised
+/// an underflow in one of the previus try_sub operation.
+pub(crate) const EEXPECTED_UNDERFLOW: u64 = 0x02_0004;
+
+pub(crate) const ECODE_INVARIANT_BROKEN: u64 = 0x02_0005;
+
+#[derive(Debug)]
+pub enum BoundedMathError {
+    Overflow,
```

Comment:
We have 5 error codes Bound_underflow, Bound_overflow, Expected_underflow, Expected_overflow, Code_invariant_Broken. We have only two variants for BoundedMathError?

Reply:
correct.

addition and subtraction operations can only fail with overflow and underflow. Caller of those functions decides what is expected, what isn't, and what error with which context should be propagated to the caller.